packages/playwright-core/src/server/dispatchers/elementHandlerDispatcher.ts

```diff
@@ -217,18 +221,10 @@ export class ElementHandleDispatcher extends JSHandleDispatcher implements chann
   }
 
   private _browserContextDispatcher(): BrowserContextDispatcher {
-    const scope = this.parentScope();
-    if (scope instanceof BrowserContextDispatcher)
-      return scope;
-    if (scope instanceof PageDispatcher)
-      return scope.parentScope();
-    if ((scope instanceof WorkerDispatcher) || (scope instanceof FrameDispatcher))  {
-      const parentScope = scope.parentScope();
-      if (parentScope instanceof BrowserContextDispatcher)
-        return parentScope;
-      return parentScope.parentScope();
-    }
-    throw new Error('ElementHandle belongs to unexpected scope');
+    const parentScope = this.parentScope().parentScope();
+    if (parentScope instanceof BrowserContextDispatcher)
```

Comment:
But this is a change from 

```js
    const scope = this.parentScope();
    if (scope instanceof BrowserContextDispatcher)
      return scope;
```

Reply:
This is because `ElementHandleDispatcher` can only be a child of `FrameDispatcher`, so this code is simplified.